### ISSUE TITLE
Batch alerts by data provider

### ISSUE THREAD

**Issue by turley85** [AtlasOfLivingAustralia/alerts#267]
We recently had a new dataset loaded, which has resulted in hundreds of alert taxa:
<img width="1257" alt="image" src="https://github.com/user-attachments/assets/41f0dbb9-3dd0-4b95-a9bd-e8c7f9567445">

This will be difficult for users to sort though. To improve UX, could we:

1. Add subheading and batch alerts by data supplier (Eg. all alerts from iNaturalist under the subheading "iNaturalist" in the email)
2. Sort alerts such that the data supplier with least number of alerts appears at the top of each email. 

Thanks

**Comment by qifeng-bai** [comment]
<img width="749" height="281" alt="Image" src="https://github.com/user-attachments/assets/180f2bfd-f557-4cb8-8d30-0548c94d9a90" />

**Comment by qifeng-bai** [comment]
Added the data provider name and record count at the top.
Provided an internal anchor link or each data provider. Clicking the link will jump to the first record for that data provider.
**Note: The internal href anchor links may not work in some email systems, such as Gmail and Outlook.**

<img width="872" height="809" alt="Image" src="https://github.com/user-attachments/assets/02eeceae-5b76-46cd-b2dd-bfa315364234" />

**Comment by kylie-m** [comment]
Thanks for your work on this Bai! I'll include it in my testing this week and let you know how it goes, and if I have any UX/UI feedback :)

**Comment by qifeng-bai** [comment]
@kylie-m It is not included in this release, but it is worth mentioning in UX/UI meeting

**Comment by turley85** [comment]
Thanks @qifeng-bai!

**Comment by qifeng-bai** [comment]
todo: doublecheck the sensitive data process. Vaguely remember we hide locations for the sensitive data

**Comment by kylie-m** [comment]
Sensitive locations should be obfuscated in csv downloads (biocache - unless users have a relevant ROLE_SDS_xxxxxx) and also in the UI in biocache. Same for the alerts, unless we know the user receiving it has an appropriate SDS role (so perhaps it's better to obfuscate for all?)

For testing, here are some species that should be obfuscated:

**Comment by qifeng-bai** [comment]
> Sensitive locations should be obfuscated in csv downloads (biocache - unless users have a relevant ROLE_SDS_xxxxxx) and also in the UI in biocache. Same for the alerts, unless we know the user receiving it has an appropriate SDS role (so perhaps it's better to obfuscate for all?)

Seems like a big task. When emails are being sent, the email service is not able to check the roles of the recipients. Unless we completely rewrite this part to generate different records for different roles and send individual emails, it would be difficult to implement. The same applies to the CSV. Once the CSV is generated, any user with the Biosecurity Manager role can download it.

The simplest and most realistic solution may be to obfuscate the locations of sensitive records in emails, while keeping either the precise or obfuscated location in the CSV.

**Comment by kylie-m** [comment]
> The simplest and most realistic solution may be to obfuscate the locations of sensitive records in emails, while keeping either the precise or obfuscated location in the CSV.

Yes I think so too, that was what Iw as trying to say  - obfuscate in all alert emails. For the Biosecurity reporting CSV - that's ok as it's internal ALA use only

**Comment by qifeng-bai** [comment]
@Adam wrote: The buffer I mentioned was for the areas uploaded to spatial that make their way into biosecurity species list fq values. This is to ensure inclusion when coordinates are obfuscated by randomly moving the occurrence location 10km (some sds movements may be smaller or larger). This is one of the many things to keep in mind when constructing an area that will be uploaded to spatial and used as part of an fq in a biosecurity species list. No alerts work is required.
The coordinates visible in the biosecurity alerts email should be retrieved from biocache-service **without the use of any JWT**. When no JWT is used biocache-service cannot provided un-obfuscated coordinates. Very few APIs in biocache-service consume JWT, but it is best to avoid sending JWT to biocache-service unnecessarily.


@qifeng-bai  I can confirm that the token is not sent to Biocache. Check `https://github.com/AtlasOfLivingAustralia/alerts/blob/develop/grails-app/services/au/org/ala/alerts/BiosecurityService.groovy#L268`
